cjdns-snode/src/peer/info.rs

```diff
@@ -38,11 +44,18 @@ impl PeerList {
 
 impl Peer {
     fn info(&self) -> PeerInfo {
+        // todo check what is received on incoming connection
         PeerInfo {
-            addr: self.addr.clone(),
+            addr: self.get_peer_ip(),
             outstanding_requests: self.get_outstanding_reqs_count(),
             msgs_on_wire: 0, //TODO useless?
             msg_queue: 0, //self.msg_queue.len(), //TODO possible?
         }
     }
+
+    fn get_peer_ip(&self) -> String {
+        let uri = self.addr.parse::<Uri>().expect("internal error: can't create uri from peer address");
+        let ip = uri.host().expect("internal error: no host in uri, created from peer address");
+        ip.trim_start_matches("[").trim_end_matches("]").to_string()
+    }
```

Comment:
Maybe better to store in `addr` field directly what is needed here (ip address), because anyway this field is only for debugging (as far as I remember)? Probably needs a closer look.

Reply:
Done, but more files have been changed